tekton/ci/jobs/tekton-github-tasks-completed.yaml

```diff
@@ -0,0 +1,68 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: github-tasks-completed
+  namespace: tektonci
+  description: |
+    Verifies that a PR has all the tasks completed
+spec:
+  params:
+    - name: body
+      description: The body of the Pull Request
+  stepTemplate:
+    env:
+      - name: HOME
+        value: /tekton/home
+  steps:
+    - name: install-pyyaml
+      image: python:3-alpine
+      script: |
+        pip install pyyaml --user
+    - name: check-github-tasks-completed
+      image: python:3-alpine
+      script: |
```

Comment:
We should probably break this out into a dedicated `.py` with unit tests.

Reply:
that's a great suggestion but it seems to me that we haven't done that with any other ci jobs in this repo -- or is there one doing this? maybe this should be part of a larger refactor? wouldn't want this job to be a special case or inconsistent with the rest

Reply:
I'd say that's true only for jobs that are relying on script - my assumption is a majority of tools we're pulling in by image (e.g. golanglint-ci) have tests in their underlying repo. Ideally we should aim for Tasks to have the same level of testing that we have for the rest of our repos - at the end of the day we're running both in prod eventually 😅 .

This is a good signal that we're shying away from writing tests for our script based Tasks because it's difficult. Similar to the requirements.txt thread below - I suspect that this friction is mostly coming from the need to first create a docker image / the fact that scripts are embedded in yaml files, making them unusable for their language's standard tooling. I think this is a good opportunity for improvement, since I'm sure other Tekton users are falling into the same issues.

Re pushing until a larger refactor - It would be great if we can improve as we go here. Kicking the can down the road is going to create more debt to resolve later, and makes it even more tempting to say "everything else is untested, one more isn't going to hurt". That said, since part of this is looks like a tooling issue with Tekton itself, I'm okay if you want to assign yourself an issue and mark a TODO here to unblock this PR.

Reply:
I agree with refactoring out scripts and testing them. Even more, those refactored pieces of software can have their own lifecycle and versioning - this is something I've been thinking about with regards to the Catalog (https://github.com/tektoncd/catalog/issues/29). However, I wouldn't say that `we're shying away from writing tests for our script based Tasks`. It's more so that we don't have a precedence or tooling set up for that yet and I don't have the capacity to set that up right now as part of this PR. I agree with making improvements as we go, but that's practical for small improvements over time and what you're proposing seems like a big change for our tooling. The progressive improvement I'm hoping to make here is automating the pull requests checklist. I've opened an issue https://github.com/tektoncd/plumbing/issues/972 for refactoring scripts in plumbing more broadly, but can't assign myself at this time.

Reply:
SGTM. Thanks! ^_^